fabric-rendering-v1/src/main/java/net/fabricmc/fabric/mixin/client/rendering/MixinWorldRenderer.java

```diff
@@ -166,8 +167,44 @@ private void afterRender(CallbackInfo ci) {
 		WorldRenderEvents.END.invoker().onEnd(context);
 	}
 
-	@Inject(method = "Lnet/minecraft/client/render/WorldRenderer;reload()V", at = @At("HEAD"))
+	@Inject(method = "reload()V", at = @At("HEAD"))
 	private void onReload(CallbackInfo ci) {
 		InvalidateRenderStateCallback.EVENT.invoker().onInvalidate();
 	}
+
+	@Inject(at = @At("HEAD"), method = "renderWeather", cancellable = true)
+	private void renderWeather(LightmapTextureManager manager, float tickDelta, double x, double y, double z, CallbackInfo info) {
+		EnvironmentRenderers.WeatherRenderer renderer = null;
+
+		if (this.client.world != null) renderer = EnvironmentRenderers.getWeatherRenderer(world.getRegistryKey());
+
+		if (renderer != null) {
+			renderer.render(this.client, manager, tickDelta, x, y, z);
+			info.cancel();
+		}
+	}
+
+	@Inject(at = @At("HEAD"), method = "renderClouds(Lnet/minecraft/client/util/math/MatrixStack;Lnet/minecraft/util/math/Matrix4f;FDDD)V", cancellable = true)
+	private void renderCloud(MatrixStack matrices, Matrix4f matrix4f, float tickDelta, double cameraX, double cameraY, double cameraZ, CallbackInfo info) {
+		EnvironmentRenderers.CloudRenderer renderer = null;
+
+		if (this.client.world != null) renderer = EnvironmentRenderers.getCloudRenderer(world.getRegistryKey());
+
+		if (renderer != null) {
+			renderer.render(this.client, matrices, matrix4f, tickDelta, cameraX, cameraY, cameraZ);
+			info.cancel();
+		}
+	}
+
+	@Inject(at = @At("HEAD"), method = "renderSky", cancellable = true)
+	private void renderSky(MatrixStack matrices, Matrix4f matrix4f, float tickDelta, Runnable runnable, CallbackInfo info) {
+		EnvironmentRenderers.SkyRenderer renderer = null;
+
+		if (this.client.world != null) renderer = EnvironmentRenderers.getSkyRenderer(world.getRegistryKey());
```

Comment:
Formatting.
How does this pass checkstyle???

Reply:
It allows single-line `if`s and such:
https://github.com/FabricMC/fabric/blob/46582230fb580d4c1f71e4b0737df27417ec9cb1/checkstyle.xml#L91-L94